akka-docs/src/main/paradox/stream/operators/Source-or-Flow/monitor.md

```diff
@@ -14,7 +14,35 @@ Materializes to a `FlowMonitor` that monitors messages flowing through or comple
 
 Materializes to a `FlowMonitor` that monitors messages flowing through or completion of the operators. The operators otherwise
 passes through elements unchanged. Note that the `FlowMonitor` inserts a memory barrier every time it processes an
-event, and may therefore affect performance.
+event, and may therefore affect performance. The provided `FlowMonitor` contains a `state` field you can use to peek
+and get information about the stream. 
+
+## Example
+
+The example below uses the `monitorMat` variant of `monitor`. The only difference betwee the two operators is 
+that `monitorMat` has a `combine` argument so we can decide which materialization value to keep. In the sample 
+below be `Keep.right` so only the `FlowMonitor[Int]` is returned. 
```

Comment:
Monitor is deprecated since 2.5.17 so the docs should be the `monitorMat` page rather than this one?

Reply:
[There's](https://github.com/akka/akka/blob/21048496582d351f7bd1e061f6712bb8afa8ae38/akka-stream/src/main/scala/akka/stream/javadsl/Source.scala#L4045-L4076) `monitor()`,  `monitor(combine)` (deprecated) and `monitorMat(combine)`.

There's only a `monitor.md` and I don't think we need a `monitorMat.md`. 

I used `monitorMat` in the sample code to keep types simpler and I didn't add a new page for `monitorMat` because the difference between the two is too small. It's a bit of a mess as it is. I can remove the usage of `monitorMat` and deal with the more complex types in the sample.

Reply:
I'm trying different things and the moment I remove the `Thread.sleep` or the `.toCompletableFuture().get()` the sample becomes a bit useless.

I'll push the code as is and mark this PR draft to think about other options...

Reply:
Agh, I missed that, what a little mess! :D